### PR TITLE
Kp maintenance

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #89 by adding new tags for the updated GFZ Kp data products.  The prior tag ('') has been deprecated and scheduled for removal. 

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

Updated unit tests.  Also:

```
import datetime as dt
import pysat
import pysatSpaceWeather as pysat_sw

# Initialize the two new Instrument combos
kp_def = pysat.Instrument('sw', 'kp', tag='def')
kp_now = pysat.Instrument('sw', 'kp', tag='now')

# Get the data
kp_def.download(start=dt.datetime(2022, 1, 1))
kp_now.download(start=dt.datetime(2022, 1, 1))

# Load the data
kp_def.load(date=dt.datetime(2022, 1, 1), use_header=True)
kp_now.load(date=dt.datetime(2022, 1, 1), use_header=True)

# Check out the new references and acknowledgements
print(kp_def.acknowledgements, kp_now.references)
```

This yeilds:
```
CC BY 4.0, The Kp index was introduced by Bartels (1949) and is produced by Geomagnetic Observatory Niemegk, GFZ German Research Centre for Geosciences.  Please cite the references in the 'references' attribute Matzka, J., Bronkalla, O., Tornow, K., Elger, K. and Stolle, C., 2021. Geomagnetic Kp index. V. 1.0. GFZ Data Services, doi:10.5880/Kp.0001
Matzka, J., Stolle, C., Yamazaki, Y., Bronkalla, O. and Morschhauser, A., 2021. The geomagnetic Kp index and derived indices of geomagnetic activity. Space Weather,doi:10.1029/2020SW002641
```

Now, show the nowcast and definitive indices can differ `kp_def['Kp'] == kp_now['Kp']`:
```
2022-01-01 00:00:00     True
2022-01-01 03:00:00     True
2022-01-01 06:00:00    False
2022-01-01 09:00:00     True
2022-01-01 12:00:00     True
2022-01-01 15:00:00     True
2022-01-01 18:00:00     True
2022-01-01 21:00:00    False
Name: Kp, dtype: bool
```

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.8
* Any details about your local setup that are relevant: develop branch of pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [N/A] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysatSpaceWeather/instruments/methods/kp_ap.py [hunk 1]
```diff
@@ -212,16 +247,15 @@ def calc_daily_Ap(ap_inst, ap_name='3hr_ap', daily_name='Ap',
...
-                     '24-h running average of 3-hourly ap indices'}
-
-        ap_inst.meta[running_name] = meta_dict
+        initialize_ap_metadata(ap_inst.meta, running_name,
+                               ap_inst.meta[ap_name][
+                                   ap_inst.meta.labels.fill_val])
+        ap_inst.meta[running_name,
+                     ap_inst.meta.labels.desc] = "running daily Ap index"
+        ap_inst.meta[running_name,
+                     ap_inst.meta.labels.notes] = ''.join(['24-h running ave',
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/methods/kp_ap.py:256]:
```suggestion
                     ap_inst.meta.labels.notes] = ''.join(['24-h running average',
```

File: pysatSpaceWeather/instruments/sw_kp.py [hunk 2]
```diff
@@ -352,63 +459,109 @@ def download(date_array, tag, inst_id, data_path):
...
-                        # then continue on
-                        os.remove(saved_fname)
-                        logger.info(' '.join(('File not available for',
-                                              dl_date.strftime('%x'))))
-
-        ftp.close()
+                logger.info(' '.join(('Downloading file for',
+                                      dl_date.strftime('%Y'))))
+                furl = ''.join([burl, fname])
+                req = requests.get(furl)
```
Comment (rstoneback) [inline pysatSpaceWeather/instruments/sw_kp.py:494]: I'm guessing that requests handles things more gracefully on its own, thus the try is no longer needed?

No feedback if the file is not available for download?

**Review by rstoneback — COMMENTED** [review]
I was able to run the PR opening code without issue. 😄

File: pysatSpaceWeather/instruments/sw_kp.py (hunk 2 above)
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_kp.py:494]: The try was from the ftp, the http stuff is much simpler code.  As seen elsewhere, you'll get a JSON error if trying to download a file that doesn't exist.  But that is treated as an acceptable error in most places.

File: pysatSpaceWeather/instruments/methods/kp_ap.py (hunk 1 above)
Comment (aburrell) [reply pysatSpaceWeather/instruments/methods/kp_ap.py:256]: I shortened this for the line length.  How much to you not want average abbreviated?

File: pysatSpaceWeather/instruments/sw_kp.py (hunk 2 above)
Comment (rstoneback) [reply pysatSpaceWeather/instruments/sw_kp.py:494]: So a JSON error pops up but the code continues to run? If so, then I'm fine.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**